extension/dynamicconfig/service/localconfig.go

```diff
@@ -0,0 +1,115 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/spf13/viper"
+	"go.opentelemetry.io/collector/config"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/dynamicconfig/model"
+	pb "github.com/vmingchen/opentelemetry-proto/gen/go/collector/dynamicconfig/v1"
+)
+
+// LocalConfigBackend is a ConfigBackend that uses a local file to determine
+// what schedules to change. The file is read live, so changes to it will
+// reflect immediately in the configs.
+type LocalConfigBackend struct {
+	viper        *viper.Viper
+	MetricConfig *model.MetricConfig
```

Comment:
Sorry if I missed it: is there a reason this field needs to be public?

Reply:
Great question, I decided to make it public to grant easy access for outsiders to use, but in hindsight it does seem unnecessarily open, especially considering the race-conditions possible. Making it private